core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -157,6 +161,15 @@ class HistoryServer(
     contextHandler.setContextPath(HistoryServer.UI_PATH_PREFIX)
     contextHandler.addServlet(new ServletHolder(loaderServlet), "/*")
     attachHandler(contextHandler)
+
+    // Register history server source to history server metrics system.
+    cacheMetrics.init()
```

Comment:
why are we calling init here?  Doesn't this end up getting called twice?

Reply:
> why are we calling init here? Doesn't this end up getting called twice?

Only called once here, before this method is never called. I am also confused about this..